packages/devtools_app/lib/src/flutter/banner_messages.dart

```diff
@@ -0,0 +1,304 @@
+          children: messagesToShow,
+        );
+      }
+    }
+    return const SizedBox();
+  }
+}
+
+class BannerMessage extends StatelessWidget implements UniqueMessage {
+  const BannerMessage({
+    @required this.messageId,
+    @required this.textSpans,
+    @required this.backgroundColor,
+    @required this.foregroundColor,
+  });
+
+  final String messageId;
+  final List<TextSpan> textSpans;
+  final Color backgroundColor;
+  final Color foregroundColor;
+
+  @override
+  String get id => messageId;
+
+  @override
+  Widget build(BuildContext context) {
+    final bannerMessagesController = Controllers.of(context).bannerMessages;
+    return Card(
+      color: backgroundColor,
+      child: Padding(
+        padding: const EdgeInsets.all(defaultSpacing),
+        child: Row(
+          children: <Widget>[
+            Expanded(
+              child: Container(
+                child: RichText(
+                  text: TextSpan(
+                    children: textSpans,
+                  ),
+                ),
+              ),
+            ),
+            const SizedBox(width: denseSpacing),
+            CircularIconButton(
+              icon: Icons.close,
+              backgroundColor: backgroundColor,
+              foregroundColor: foregroundColor,
+              // TODO(kenz): animate the removal of this message.
+              onPressed: () => bannerMessagesController.dismissMessage(id),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _BannerError extends BannerMessage {
+  const _BannerError({
+    @required String id,
+    @required List<TextSpan> textSpans,
+    @required DevToolsScreenType screenType,
+  }) : super(
+          messageId: id,
+          textSpans: textSpans,
+          backgroundColor: devtoolsError,
+          foregroundColor: foreground,
+        );
+
+  static const foreground = Colors.white;
+  static const linkColor = Color(0xFF88D7FC);
+}
+
+// TODO(kenz): add "Do not show this again" option to warnings.
+class _BannerWarning extends BannerMessage {
+  const _BannerWarning({
+    @required String id,
+    @required List<TextSpan> textSpans,
+    @required DevToolsScreenType screenType,
+  }) : super(
+          messageId: id,
+          textSpans: textSpans,
+          backgroundColor: devtoolsWarning,
+          foregroundColor: foreground,
+        );
+
+  static const foreground = Colors.black87;
+  static const linkColor = Color(0xFF055BF0);
+}
+
+class DebugModePerformanceMessage extends StatelessWidget
+    implements UniqueMessage {
+  const DebugModePerformanceMessage(this.screenType);
+
+  final DevToolsScreenType screenType;
+
+  @override
+  String get id => 'DebugModePerformanceMessage - $screenType';
+
+  @override
+  Widget build(BuildContext context) {
+    return _BannerError(
+      id: id,
+      textSpans: [
+        const TextSpan(
+          text:
+              'You are running your app in debug mode. Debug mode performance '
```

Comment:
nit: this is probably cleaner as a ''' literal. You wouldn't need to escape anything.

Reply:
done